src/tasks/NetTraceToMibcConverterTask/NetTraceToMibcConverter.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Text.Json.Serialization;
+
+public class NetTraceToMibcConverter : Microsoft.Build.Utilities.Task
+{
+
+    /// <summary>
+    /// Path to the mibc converter tool
+    /// </summary>
+    public string? MibcConverterBinaryPath { get; set; }
+
+    /// <summary>
+    /// Entries for assemblies referenced in a .nettrace file. Important when you run traces against an executable on a different machine / device
+    /// </summary>
+    [Required]
+    public ITaskItem[] Assemblies { get; set; } = Array.Empty<ITaskItem>();
+
+    /// <summary>
+    /// NetTrace file to use when invoking dotnet-pgo for
+    /// </summary>
+    [Required]
+    public string NetTracePath { get; set; } = "";
+
+    /// <summary>
+    /// File used to track hashes of assemblies, to act as a cache
+    /// Output files don't get written, if they haven't changed
+    /// </summary>
+    public string? CacheFilePath { get; set; }
+
+    /// <summary>
+    /// Directory where the mibc file will be placed
+    /// </summary>
+    [NotNull]
+    [Required]
+    public string? OutputDir { get; set; }
+
+    /// <summary>
+    /// The path to the mibc file generated from the converter.
+    /// </summary>
+    [Output]
+    public string? MibcProfilePath { get; set; }
+
+    private bool ProcessAndValidateArguments()
+    {
+        if (!File.Exists(MibcConverterBinaryPath))
+        {
+            Log.LogError($"{nameof(MibcConverterBinaryPath)}='{MibcConverterBinaryPath}' doesn't exist.");
+            return false;
+        }
+
+        if (Assemblies.Length == 0)
+        {
+            Log.LogError($"'{nameof(Assemblies)}' is required.");
+            return false;
+        }
+
+        if (!File.Exists(NetTracePath))
+        {
+            Log.LogError($"{nameof(NetTracePath)}='{NetTracePath}' doesn't exist");
+            return false;
+        }
+
+        foreach (var asmItem in Assemblies)
+        {
+            string? fullPath = asmItem.GetMetadata("FullPath");
+            if (!File.Exists(fullPath))
+                throw new LogAsErrorException($"Could not find {fullPath} to AOT");
+        }
+
+        return !Log.HasLoggedErrors;
+    }
+
+    public override bool Execute()
+    {
+        try
+        {
+            return ExecuteInternal();
+        }
+        catch (LogAsErrorException laee)
+        {
+            Log.LogError(laee.Message);
+            return false;
+        }
+    }
+
+    private bool ExecuteInternal()
+    {
+        if (!ProcessAndValidateArguments())
+            return false;
+
+        if (!ProcessNettrace(NetTracePath))
+            return false;
+
+        return !Log.HasLoggedErrors;
+    }
+
+    private bool ProcessNettrace(string netTraceFile)
+    {
+        var outputMibcPath = Path.Combine(OutputDir, Path.ChangeExtension(Path.GetFileName(netTraceFile), ".mibc"));
+
+        StringBuilder pgoArgsStr = new StringBuilder(string.Empty);
```

Comment:
If we want to not construe this with pgo (like how it was `PgoBinaryPath` -> `MibcConverterBinaryPath` and named `NetTraceToMibcConverter`), then should we rename this to `mibcConverterArgsStr`?